comp/api/api/apiimpl/server.go

```diff
@@ -47,43 +48,23 @@ func (server *apiServer) startServers() error {
 		return fmt.Errorf("unable to get IPC address and port: %v", err)
 	}
 
-	additionalHostIdentities := []string{apiAddr}
-
-	ipcServerHost, ipcServerHostPort, ipcServerEnabled := getIPCServerAddressPort()
-	if ipcServerEnabled {
-		additionalHostIdentities = append(additionalHostIdentities, ipcServerHost)
-	}
-
-	tlsKeyPair, tlsCertPool, err := initializeTLS(additionalHostIdentities...)
-	if err != nil {
-		return fmt.Errorf("unable to initialize TLS: %v", err)
-	}
-
-	// tls.Config is written to when serving, so it has to be cloned for each server
-	tlsConfig := func() *tls.Config {
-		return &tls.Config{
-			Certificates: []tls.Certificate{*tlsKeyPair},
-			NextProtos:   []string{"h2"},
-			MinVersion:   tls.VersionTLS12,
-		}
-	}
-
 	tmf := observability.NewTelemetryMiddlewareFactory(server.telemetry)
 
 	// start the CMD server
 	if err := server.startCMDServer(
 		apiAddr,
-		tlsConfig(),
-		tlsCertPool,
 		tmf,
 		server.cfg,
 	); err != nil {
 		return fmt.Errorf("unable to start CMD API server: %v", err)
 	}
 
 	// start the IPC server
-	if ipcServerEnabled {
-		if err := server.startIPCServer(ipcServerHostPort, tlsConfig(), tmf); err != nil {
+	if ipcServerPort := server.cfg.GetInt("agent_ipc.port"); ipcServerPort != 0 {
```

Comment:
What happen if `ipcServerPort == 0` ?

Reply:
if `ipcServerPort` is set to 0 we consider that it should not be started.